VolumeLogic/VolumeMixer.py

```diff
@@ -10,18 +10,40 @@ class VolumeMixer(IObserver, Subject):
         ChangedValue.COLOR: lambda application, value: application.color_matrix(value),
     }
 
-    def __init__(self):
+    def __init__(self, os_connection):
         super().__init__()
+        self.os_connection = os_connection
+        Application.os_connection = os_connection
         self.applications = [None] * self.num_of_knobs
-        self.modified_index = 0
 
     def update(self, subject, arg):
         for index, application in enumerate(self.applications):
             if application == subject:
-                self.modified_index = index
-                self.notify_all(arg)
+                self.notify_all((index, arg))
                 break
 
+    @property
+    def applications(self):
+        return self.applications
+
+    @applications.setter
+    def applications(self, applications):
+        self.applications = applications
+
+    def get_running_applications(self):
+        """
+        Gets all the applications currently running on the computer.
+
+        :return: List of applications from the computer.
+        """
+        applications = []
+
+        processes = self.os_connection.getRunningProcesses()
```

Comment:
Is this the only reference to `self.os_connection`? I wonder if we could avoid having an extra instance of it.

Reply:
There are some more references in Application, and when we add in play/pause, etc there will be another reference in this class